torch/_dynamo/variables/builtin.py

```diff
@@ -1375,6 +1375,11 @@ def _comparison(self, tx, left, right):
         def _unimplemented():
             unimplemented(f"comparison {typestr(left)} {op} {typestr(right)}")
 
+        if op.__name__ == "is_":
+            # If the two objects are of different type, we can safely return False
```

Comment:
This isn't testing the Python type, this is testing the VT type

Reply:
not good enough. see: https://github.com/pytorch/pytorch/issues/111713